Automattic/wp-calypso
# Checkout: Remove edit button and show only longer term variants

#### Proposed Changes

Monthly plan variants are being selected too frequently in checkout. To help with this, the term variant picker (for WordPress.com, not Jetpack) was hidden behind an "Edit" button in https://github.com/Automattic/wp-calypso/pull/68606. However, now longer term variants are not being selected often enough.

This PR removes the "Edit" button and alters the variant picker (again, for WordPress.com and not Jetpack since Jetpack has no other way to select a monthly plan) so that it will not display variants which have a shorter term than the item that was in the cart when checkout loaded. 

Fixes https://github.com/Automattic/wp-calypso/issues/69633

Note: there's also a plan to consider switching to radio buttons from the designs in https://github.com/Automattic/wp-calypso/pull/69494#issuecomment-1295315829 (we can reuse some of the code in https://github.com/Automattic/wp-calypso/pull/66027) but that will be done separately.

#### Screenshots

Before:
<img width="564" alt="Screen Shot 2022-11-02 at 6 21 35 PM" src="https://user-images.githubusercontent.com/2036909/199613821-1ba9f122-ab53-4708-aaba-0105395e1aca.png">

After:
<img width="568" alt="Screen Shot 2022-11-02 at 6 22 11 PM" src="https://user-images.githubusercontent.com/2036909/199613846-b881062b-0314-44da-9bbd-22317352e662.png">

Before:
<img width="567" alt="Screen Shot 2022-11-02 at 6 23 01 PM" src="https://user-images.githubusercontent.com/2036909/199613910-8e499cb0-542e-49c9-8295-745caf8730ea.png">

After:
<img width="567" alt="Screen Shot 2022-11-02 at 6 23 08 PM" src="https://user-images.githubusercontent.com/2036909/199613920-1bec1e7c-9084-454a-a200-efe9d5c260b8.png">

#### Testing Instructions

Unit tests are included but you can test manually by doing the following.

Test annual/biennial plans:

- Add a WordPress.com annual plan to your cart and visit checkout.
- Verify that underneath the plan is a variant picker (and no "Edit" button).
- Click on the variant picker and verify that you can see the annual option and the biennial option, but not the monthly option.
- Click to select the biennial option and wait for the cart to finish updating.
- Click on the variant picker and verify that you can _still_ see the annual option and the biennial option, but not the monthly option.
- Reload the checkout page.
- Verify that there is no variant picker visible.

Test monthly plans:

- Add a WordPress.com monthly plan to your cart and visit checkout.
- Verify that underneath the plan is a variant picker (and no "Edit" button).
- Click on the variant picker and verify that you can see the monthly, annual, and biennial options.
- Click to select the annual option and wait for the cart to finish updating.
- Click on the variant picker and verify that you can _still_ see the monthly, annual, and biennial options.
- Reload the checkout page.
- Click on the variant picker and verify that you can only the annual and biennial options, but not the monthly option.

Test Jetpack plans:

- Add a Jetpack annual plan to your cart (for a Jetpack site) and visit checkout.
- Verify that underneath the plan is a variant picker (and no "Edit" button).
- Click on the variant picker and verify that you can see the monthly, annual, and biennial options.
